doc/learn/syntax-trees-with-typescript.md

```diff
@@ -0,0 +1,523 @@
+A full list of node types can be found in the [MDAST documentation](https://github.com/syntax-tree/mdast#readme).
+[Typings for MDAST are available on npm](https://www.npmjs.com/package/@types/mdast) can be installed with a package manager.
+
+```bash
+npm install --save-dev @types/mdast
+```
+
+then import the types into a TypeScript file with:
+
+```ts
+import type {Heading, Code, Link} from 'mdast'
+```
+
+or into a [JSDoc TypeScript](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html) project with:
+
+```js
+/**
+ * @typedef {import("mdast").Heading} Heading
+ * @typedef {import("mdast").Code} Code
+ * @typedef {import("mdast").Link} Link
+ */
+```
+
+#### HAST (HTML)
+
+The [**H**ypertext **A**bstract **S**yntax **T**ree (HAST)](https://github.com/syntax-tree/hast#readme) extends `unist` with types specific for HTML such as [`Element`](https://github.com/syntax-tree/hast#element), [`Comment`](https://github.com/syntax-tree/hast#comment), [`DocType`](https://github.com/syntax-tree/hast#doctype), and many more.
+A full list of node types can be found in the [HAST documentation](https://github.com/syntax-tree/hast#readme).
+[Typings for HAST are available on npm](https://www.npmjs.com/package/@types/hast) can be installed with a package manager.
+
+```bash
+npm install --save-dev @types/hast
+```
+
+then import the types into a TypeScript file with:
+
+```ts
+import type {Element, Comment, DocType} from 'hast'
+```
+
+or into a [JSDoc TypeScript](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html) project with:
+
+```js
+/**
+ * @typedef {import("hast").Element} Element
+ * @typedef {import("hast").Comment} Comment
+ * @typedef {import("hast").DocType} DocType
+ */
+```
+
+#### XAST (XML)
+
+The [E**x**tensible **A**bstract **S**yntax **T**ree (XAST)](https://github.com/syntax-tree/xast#readme) extends `unist` with types specific for XML such as [`Element`](https://github.com/syntax-tree/xast#element), [`CData`](https://github.com/syntax-tree/xast#cdata), [`Instruction`](https://github.com/syntax-tree/xast#instruction), and many more.
+A full list of node types can be found in the [XAST documentation](https://github.com/syntax-tree/xast#readme).
+[Typings for XAST are available on npm](https://www.npmjs.com/package/@types/xast) can be installed with a package manager.
+
+```bash
+npm install --save-dev @types/xast
+```
+
+then import the types into a TypeScript file with:
+
+```ts
+import type {Element, CData, Instruction} from 'xast'
+```
+
+or into a [JSDoc TypeScript](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html) project with:
+
+```js
+/**
+ * @typedef {import("xast").Element} Element
+ * @typedef {import("xast").CData} CData
+ * @typedef {import("xast").Instruction} Instruction
+ */
+```
+
+### How to traverse a syntax tree
```

Comment:
Make this section more specific to typescript. There is already an in-depth article on tree traversal. So cross link to and from that: https://github.com/unifiedjs/unifiedjs.github.io/blob/main/doc/learn/tree-traversal.md

Reply:
linked to tree traversal guide